sklearn/utils/testing.py

```diff
@@ -136,7 +136,8 @@ def assert_warns(warning_class, func, *args, **kw):
             raise AssertionError("No warning raised when calling %s"
                                  % func.__name__)
 
-        found = any(warning.category is warning_class for warning in w)
+        found = any(issubclass(warning.category, warning_class)
```

Comment:
This is another controversial change which requires a design decision. Despite a similar name, `sklearn.utils.assert_warns_message` works in a different way than `sklearn.utils.assert_warns` - it checks for issubclass, not for exact match of classes. So I had two options:

1. Change `sklearn.utils.assert_warns` to use issubclass - it should fix tests.
2. Change `sklearn.utils.assert_warns_message` to not use `is`, then fix tests by expecting SklearnDeprecationWarnings (it may also require some other changes, I haven't checked that).

I liked (1) more because

* issubclass is consistent with how warning filters work and with examples in Python docs;
* (2) change is more invasive, and it could break sklearn user's tests because `assert_warns_message` becomes more strict.